netdata/netdata
# Update libbpf 1.0.1

##### Summary
More details after tests...

##### Test Plan

1. Compile this branch
2. Start netdata
3. Verify that we do not have `libbpf` issues:
```sh
$ grep libbpf /var/log/netdata/error.log
```
4. Take a look at chart http://localhost:19999/api/v1/data?chart=netdata.ebpf_load_methods&after=-10 to see threads running.

##### Additional Information
This PR was tested on:

| Linux Distribution | kernel version | libbpf  | Threads |
|-------------------------|--------------------|----------|-------------|
| Slackware Current  |     5.19.12     | [slackware_5_19_logs.txt](https://github.com/netdata/netdata/files/9727659/slackware_5_19_logs.txt)        | [slackware_5_19_threads.txt](https://github.com/netdata/netdata/files/9727660/slackware_5_19_threads.txt) | 
| Arch Linux | 5.19.13-arch1-1 | [arch_linux_5_19_logs.txt](https://github.com/netdata/netdata/files/9728623/arch_linux_5_19_logs.txt) | [arch_linux_5_19_threads.txt](https://github.com/netdata/netdata/files/9728624/arch_linux_5_19_threads.txt) | 
| Ubuntu 22.04 |  5.15.0-33-generic |  [Ubuntu_5_15_logs.txt](https://github.com/netdata/netdata/files/9728864/Ubuntu_5_15_logs.txt) | [Ubuntu_5_15_threads.txt](https://github.com/netdata/netdata/files/9728865/Ubuntu_5_15_threads.txt) |
| Alma 9 | 5.14.0-70.26.1.el9_0.x86_64 |  [alma_5_14_log.txt](https://github.com/netdata/netdata/files/9728974/alma_5_14_log.txt) | [alma_5_14_threads.txt](https://github.com/netdata/netdata/files/9728975/alma_5_14_threads.txt) |
| Debian 11 | 5.10.140-1 | [debian_5_10_logs.txt](https://github.com/netdata/netdata/files/9729058/debian_5_10_logs.txt) | [debian_5_10_threads.txt](https://github.com/netdata/netdata/files/9729059/debian_5_10_threads.txt) |
| Slackware Current  |    5.4.210    |  [slackware_5_4_logs.txt](https://github.com/netdata/netdata/files/9729874/slackware_5_4_logs.txt) | [slackware_5_4_threads.txt](https://github.com/netdata/netdata/files/9729875/slackware_5_4_threads.txt) |
| Alma 8.6           | 4.18.0-372.26.1.el8_6.x86_64 | [alma_4_18_logs.txt](https://github.com/netdata/netdata/files/9735130/alma_4_18_logs.txt) | [alma_4_18_threads.txt](https://github.com/netdata/netdata/files/9735131/alma_4_18_threads.txt) | 
| Ubuntu 18.04 | 4.15.0-180-generic | [ubuntu_18_04_logs.txt](https://github.com/netdata/netdata/files/9736961/ubuntu_18_04_logs.txt) | [ubuntu_18_04_threads.txt](https://github.com/netdata/netdata/files/9736962/ubuntu_18_04_threads.txt) | 
| Slackware Current | 4.14.290 | [slackware_4_14_logs.txt](https://github.com/netdata/netdata/files/9737123/slackware_4_14_logs.txt) | [slackware_4_14_threads.txt](https://github.com/netdata/netdata/files/9737125/slackware_4_14_threads.txt) | 
| CentOS 7.9 | 3.10.0-1160.71.1.el7.x86_64. | [centos_3_10_logs.txt](https://github.com/netdata/netdata/files/9737423/centos_3_10_logs.txt) | [centos_3_10_threads.txt](https://github.com/netdata/netdata/files/9737424/centos_3_10_threads.txt) |


<details> <summary>For users: How does this change affect me?</summary>
Describe the PR affects users: 
- Which area of Netdata is affected by the change? ebpf.plugin
- Can they see the change or is it an under the hood? If they can see it, where? This does not affect user visualization. It brings update for `libbpf`
- How is the user impacted by the change? They will have a plugin more stable.
- What are there any benefits of the change? We are running latest version of libraries. 
</details>
